Source/WebCore/platform/ScrollAnimator.cpp

```diff
@@ -287,6 +287,7 @@ void ScrollAnimator::willStartAnimatedScroll()
 void ScrollAnimator::didStopAnimatedScroll()
 {
     m_scrollableArea.setScrollAnimationStatus(ScrollAnimationStatus::NotAnimating);
+    m_scrollableArea.animatedScrollDidEnd();
```

Comment:
Seems like `setScrollAnimationStatus()` could take care of calling `animatedScrollDidEnd()`

Reply:
I think we only want to call this here since this is the call for specifically an animated scroll finishing.